acceptance-tests/src/main/java/io/blueocean/ath/SmartWebElement.java

```diff
@@ -16,45 +15,26 @@
 import org.openqa.selenium.support.ui.FluentWait;
 
 import java.util.List;
-import java.util.concurrent.TimeUnit;
 
 /**
- * Wrapper around an underlying WebDriver that automatically handles waits and common gotchas
- * within blueocean.
- *
- * Accepts expressions for css and xpath, if the provided lookup starts with a /, XPath is used
+ * Wrapper around an underlying WebDriver that automatically handles waits and common gotchas.
  */
 public class SmartWebElement implements WebElement {
     private static Logger logger = Logger.getLogger(SmartWebElement.class);
-    public static final int DEFAULT_TIMEOUT = Integer.getInteger("webDriverDefaultTimeout", 3000);
     public static final int RETRY_COUNT = 3;
 
     private WebDriver driver;
+    protected WebElement element;
```

Comment:
Let me see if I can reintroduce that lazy logic so we don't hold a ref to any WebElement.

Reply:
Now I finally understand the motivation, so you can wrap the other returned `WebElement`s with this same logic. Ok, let me stew on this for just a little while

Reply:
Right. Otherwise can't really chain `findElement` calls together because we'll be returning a mix of `SmartWebElement` and `RemoteWebElement` which I think would lead to a lot of flake since some `WebElement`'s have wait behavior baked in and others do not.

The alternative with original code is that we'd need to do all our finds through `WebDriverMixin.find` to ensure we always get a SWE, which could lead to really lengthy selectors:

`find(".favorites-card-stack .pipeline-card[data-fullname="cliffs-pipeline"] .Checkbox.Favorite > label").click()`

 rather than

```
find(".favorites-card-stack .pipeline-card[data-fullname="cliffs-pipeline"]
   .findElement(".Checkbox.Favorite > label")
   .click()
```

I find the latter really useful because we can write individual methods on our page objects to grab specific elements, and we can compose those methods just by chaining the `findElement` calls.

I wonder if we could add some exception handling into SWE for `StaleElementException` that retries the locator. It wouldn't necessarily work for elements fetched via `findMany` since the expression stored to fetch them would match multiple elements. I can't say I've run into any `StaleElementException` yet but that's part of why I am working on a branch derived from this one to add really thorough tests to personalization as a testing ground.